selene-lib/src/rules/high_cyclomatic_complexity.rs

```diff
@@ -0,0 +1,423 @@
+                }
+
+                complexity
+            }
+
+            ast::Value::ParenthesesExpression(paren_expression) => {
+                count_expression_complexity(paren_expression, complexity)
+            }
+
+            ast::Value::Number(_) => complexity,
+            ast::Value::String(_) => complexity,
+            ast::Value::Symbol(_) => complexity,
+
+            ast::Value::TableConstructor(table) => count_table_complexity(table, complexity),
+
+            ast::Value::Var(ast::Var::Expression(var_expression)) => {
+                for suffix in var_expression.suffixes() {
+                    complexity = count_suffix_complexity(suffix, complexity)
+                }
+                complexity
+            }
+
+            ast::Value::Var(ast::Var::Name(_)) => complexity,
+
+            #[cfg(feature = "roblox")]
+            ast::Value::IfExpression(if_expression) => {
+                complexity += 1;
+                if let Some(else_if_expressions) = if_expression.else_if_expressions() {
+                    for else_if_expression in else_if_expressions {
+                        complexity += 1;
+                        complexity = count_expression_complexity(
+                            else_if_expression.expression(),
+                            complexity,
+                        );
+                    }
+                }
+                complexity
+            }
+
+            _ => complexity,
+        },
+
+        _ => complexity,
+    }
+}
+
+fn count_block_complexity(block: &ast::Block, starting_complexity: u16) -> u16 {
+    let mut complexity = starting_complexity;
+
+    // we don't immediately return from the matched blocks so that we can add in any complexity from the last statement
+    for statement in block.stmts() {
+        #[cfg_attr(
+            feature = "force_exhaustive_checks",
+            deny(non_exhaustive_omitted_patterns)
+        )]
+        match statement {
+            ast::Stmt::Assignment(assignment) => {
+                for expression in assignment.expressions() {
+                    complexity = count_expression_complexity(expression, complexity);
+                }
+            }
+
+            ast::Stmt::Do(do_) => {
+                complexity = count_block_complexity(do_.block(), complexity);
+            }
+
+            ast::Stmt::FunctionCall(call) => {
+                for suffix in call.suffixes() {
+                    complexity = count_suffix_complexity(suffix, complexity)
+                }
+            }
+
+            // visit_function_declaration already tracks this
+            ast::Stmt::FunctionDeclaration(_) => {}
+
+            ast::Stmt::GenericFor(generic_for) => {
+                complexity += 1;
+                for expression in generic_for.expressions() {
+                    complexity = count_expression_complexity(expression, complexity);
+                    complexity = count_block_complexity(generic_for.block(), complexity);
```

Comment:
Is block complexity meant to be computed for every single expression here? Seems like this should be outside the for loop